src/server/game/Handlers/MiscHandler.cpp

```diff
@@ -1144,3 +1144,27 @@ void WorldSession::HandleCloseInteraction(WorldPackets::Misc::CloseInteraction&
     if (_player->PlayerTalkClass->GetInteractionData().SourceGuid == closeInteraction.SourceGuid)
         _player->PlayerTalkClass->GetInteractionData().Reset();
 }
+
+void WorldSession::HandleAdventureJournalOpenQuest(WorldPackets::Misc::AdventureJournalOpenQuest& packet)
+{
+    auto adventureJournalEntry = sAdventureJournalStore[packet.AdventureJournalID];
+    if (!adventureJournalEntry)
+        return;
+
+    Quest const* quest = sObjectMgr->GetQuestTemplate(adventureJournalEntry->QuestID);
+    if (!quest)
+        return;
+
+    if (_player->HasQuest(adventureJournalEntry->QuestID) && _player->CanTakeQuest(quest, true))
+    {
+        PlayerMenu menu(_player->GetSession());
+        menu.SendQuestGiverQuestDetails(quest, _player->GetGUID(), true, false);
+    }
+}
+
+void WorldSession::HandleAdventureJournalStartQuest(WorldPackets::Misc::AdventureJournalStartQuest& packet)
+{
+    if (Quest const* quest = sObjectMgr->GetQuestTemplate(packet.QuestID))
+        if (!_player->hasQuest(packet.QuestID))
+            _player->AddQuestAndCheckCompletion(quest, nullptr);
```

Comment:
possible exploit - you gotta check whether the quest is valid for adventureguide or not (possibly some flag), would also check that in HandleAdventureJournalOpenQuest.

NEVER trust client.